### ISSUE TITLE
Missing dependencies in package.json

### ISSUE THREAD

**Issue by peterzen** [cGuille/async-polling#3]
When used in a browser environment the library depends on:

window.nodeUtil.inherits
window.nodeEventEmitter

There are dependencies are not documented in package.json and therefore the code fails when these globals are not defined.

**Comment by cGuille** [comment]
Hello @peterzen,

You closed the issue so I expect that you fixed it yourself.

I suppose you included `src/async-polling.js` in your HTML instead of the script built for the browser (in the `dist` directory).

Could you confirm this so that other users running into the same error will have a solution written in this issue?

Also, should the README provide more detailed explanations regarding this?

**Comment by peterzen** [comment]
I noticed afterwards that the `lib/` directory contains `util.js` and `event-emitter.js` which are loaded by `dist/async-polling.js`.  The issue I ran into was that the `main` entry in `package.json` points to the source file so the dependencies won't be loaded when the library is loaded via `require('async-polling')` using _webpack_ in the browser environment.  (I'm guessing browserify users run into the same issue also.)

I found the below workaround to get around the problem:

```
require('async-polling/dist/async-polling');  // built module contains dependencies

window.AsyncPolling( ...options ...); // module variable is not exported when 'window' object is present
```

I think the UMD loader would provide a more portable way to detect the runtime environment (instead of the current method of detection):

https://github.com/umdjs/umd

I hope this helps!

Thank you,

Peter
